Unhandled Exception on startup (Linux)

Building on Linux from master, starting the game loads it for a quick second after which it fails.
Text and audio look fine with dotnet core, so that seems to be fixed though with the last commits.

With `dotnet --version = 2.1.300`
```
netcoreapp2.1 [master] dotnet osu\!.dll

Unhandled Exception: osu.Framework.Allocation.RecursiveLoadException: Attempted to update dropdown to a value which wasn't contained as an item (1).
   at osu.Framework.Allocation.DependencyContainer.<>c__DisplayClass6_1.<register>b__5(Object instance)
   at System.Collections.Generic.List`1.ForEach(Action`1 action)
   at osu.Framework.Allocation.DependencyContainer.<>c__DisplayClass6_0.<register>b__1(DependencyContainer container, Object instance)
   at osu.Framework.Allocation.DependencyContainer.Inject[T](T instance, Boolean autoRegister, Boolean lazy)
   at osu.Framework.Graphics.Drawable.Load(IFrameBasedClock clock, IReadOnlyDependencyContainer dependencies)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot)
--- End of stack trace from previous location where exception was thrown ---
   at osu.Framework.Graphics.Drawable.<>c__DisplayClass15_1.<LoadAsync>b__2()
   at osu.Framework.Threading.ScheduledDelegate.RunTask()
   at osu.Framework.Threading.Scheduler.Update()
   at osu.Framework.Graphics.Drawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
   at osu.Framework.Platform.GameHost.UpdateFrame()
   at osu.Framework.Threading.GameThread.ProcessFrame()
   at osu.Framework.Threading.GameThread.runWork()
   at System.Threading.Thread.ThreadMain_ThreadStart()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
[1]    389 abort (core dumped)  dotnet osu!.dll
```

With `Mono JIT compiler version 5.12.0 (makepkg/58637d0ee7c Wed Jun  6 22:41:15 CEST 2018)`:

```
net471 [master] mono osu\!.exe
[ERROR] FATAL UNHANDLED EXCEPTION: System.InvalidOperationException: Attempted to update dropdown to a value which wasn't contained as an item (1).
  at osu.Framework.Graphics.UserInterface.Dropdown`1[T].selectionChanged (T newSelection) [0x0004c] in <b27247fbdfa243c5947e0c5577a1cd44>:0 
  at osu.Framework.Configuration.Bindable`1[T].TriggerValueChange (System.Boolean propagateToBindings) [0x00047] in <b27247fbdfa243c5947e0c5577a1cd44>:0 
  at osu.Framework.Configuration.Bindable`1[T].set_Value (T value) [0x00040] in <b27247fbdfa243c5947e0c5577a1cd44>:0 
  at osu.Framework.Configuration.Bindable`1[T].BindTo (osu.Framework.Configuration.Bindable`1[T] them) [0x00007] in <b27247fbdfa243c5947e0c5577a1cd44>:0 
  at osu.Game.Overlays.Settings.SettingsItem`1[T].set_Bindable (osu.Framework.Configuration.Bindable`1[T] value) [0x00019] in <85e8635d5e4d4fd28d04d3731eb7ac3d>:0 
  at osu.Game.Overlays.Settings.Sections.SkinSection.load (osu.Game.Configuration.OsuConfigManager config, osu.Game.Skinning.SkinManager skins) [0x000f8] in <85e8635d5e4d4fd28d04d3731eb7ac3d>:0 
  at osu.Game.Overlays.SettingsOverlay.load ()

Unhandled Exception:
System.InvalidOperationException: Attempted to update dropdown to a value which wasn't contained as an item (1).
  at osu.Framework.Graphics.UserInterface.Dropdown`1[T].selectionChanged (T newSelection) [0x0004c] in <b27247fbdfa243c5947e0c5577a1cd44>:0 
  at osu.Framework.Configuration.Bindable`1[T].TriggerValueChange (System.Boolean propagateToBindings) [0x00047] in <b27247fbdfa243c5947e0c5577a1cd44>:0 
  at osu.Framework.Configuration.Bindable`1[T].set_Value (T value) [0x00040] in <b27247fbdfa243c5947e0c5577a1cd44>:0 
  at osu.Framework.Configuration.Bindable`1[T].BindTo (osu.Framework.Configuration.Bindable`1[T] them) [0x00007] in <b27247fbdfa243c5947e0c5577a1cd44>:0 
  at osu.Game.Overlays.Settings.SettingsItem`1[T].set_Bindable (osu.Framework.Configuration.Bindable`1[T] value) [0x00019] in <85e8635d5e4d4fd28d04d3731eb7ac3d>:0 
  at osu.Game.Overlays.Settings.Sections.SkinSection.load (osu.Game.Configuration.OsuConfigManager config, osu.Game.Skinning.SkinManager skins) [0x000f8] in <85e8635d5e4d4fd28d04d3731eb7ac3d>:0 
  at osu.Game.Overlays.SettingsOverlay.load ()
```
build log: https://gist.github.com/radriaanse/5f60cc8ba5c0c0e34a737f8f979aeb99

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/60321288-unhandled-exception-on-startup-linux?utm_campaign=plugin&utm_content=tracker%2F4707269&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F4707269&utm_medium=issues&utm_source=github).
</bountysource-plugin>

--- a/osu.Game/Overlays/Settings/Sections/SkinSection.cs
+++ b/osu.Game/Overlays/Settings/Sections/SkinSection.cs
@@ -56,7 +56,13 @@ namespace osu.Game.Overlays.Settings.Sections
 
             reloadSkins();
 
-            skinDropdown.Bindable = config.GetBindable<int>(OsuSetting.Skin);
+            var skinBindable = config.GetBindable<int>(OsuSetting.Skin);
+
+            // Todo: This should not be necessary when OsuConfigManager is databased
+            if (skinDropdown.Items.All(s => s.Value != skinBindable.Value))
+                skinBindable.Value = 0;
+
+            skinDropdown.Bindable = skinBindable;
         }
 
         private void reloadSkins() => skinDropdown.Items = skins.GetAllUsableSkins().Select(s => new KeyValuePair<string, int>(s.ToString(), s.ID));

